src/renderer/widgets/CreateWallet/model/signatory-model.ts

```diff
@@ -0,0 +1,53 @@
+import { combine, createEvent, createStore, sample } from 'effector';
+
+import { SignatoryInfo } from '../lib/types';
+import { walletModel, walletUtils } from '@entities/wallet';
+import { toAccountId } from '@shared/lib/utils';
+
+const signatoriesChanged = createEvent<SignatoryInfo>();
+const signatoryDeleted = createEvent<number>();
+
+const $signatories = createStore<Map<number, Omit<SignatoryInfo, 'index'>>>(new Map([]));
+
+const $hasOwnSignatory = combine(
+  { wallets: walletModel.$wallets, signatories: $signatories },
+  ({ wallets, signatories }) =>
+    !!walletUtils.getWalletsFilteredAccounts(wallets, {
+      walletFn: (w) => !walletUtils.isWatchOnly(w) && !walletUtils.isMultisig(w),
+      accountFn: (a) => Array.from(signatories.values()).some((s) => toAccountId(s.address) === a.accountId),
+    })?.length,
+);
+
+sample({
+  clock: signatoriesChanged,
+  source: $signatories,
+  fn: (signatories, { index, name, address }) => {
+    signatories.set(index, { name, address });
```

Comment:
Modifying original store is not a good idea

Reply:
I originally thought about having a array, instead of a Map. Now the thing is that arrays have no guarantee of keeping in order (although tbh I've never had issues with it). Maps do. However, this adds a lot of complexity everywhere, since looping through Maps require `Array.from`.

Another complexity that it adds, is that Maps don't trigger a re-rendering of the stores. The only way to trigger is to create a new Map from the original one, modify it, and assign the store to the new map (a new object address).

Do you think I should rather go with arrays, and take the risk of having the array order changed?

Reply:
Right, create new map -> modify new map -> return new map

Reply:
ah got it, I made the change.